src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/illegaltype/InputIllegalTypeGenerics.java

```diff
@@ -0,0 +1,59 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.illegaltype;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Consumer;
+
+/**
+ * This test-input is intended to be checked using following configuration:
+ *
+ * illegalClassNames = {Boolean, Foo, Serializable}
+ * memberModifiers = {LITERAL_PUBLIC, FINAL}
+ *
+ */
+public abstract class InputIllegalTypeGenerics {
+
+    private Set<Boolean> privateSet; // OK
+    private java.util.List<Map<Boolean, Foo>> privateList; // OK
+    public Set<Boolean> set; // warn
+    public java.util.List<Map<Boolean, Foo>> list; // warn
+
+    private void methodCall() {
+        Bounded.<Boolean>foo(); // warn
+        final Consumer<Foo> consumer = Foo<Boolean>::foo; // warn
+    }
+
+    public <T extends Boolean, U extends Serializable> void typeParameter(T a) {} // warn
+
+    public void fullName(java.util.ArrayList<? super Boolean> a) {} // warn
+
+    public abstract Set<Boolean> shortName(Set<? super Boolean> a); // warn
+
+    public Set<? extends Foo> typeArgument() { // warn
+        return new TreeSet<Foo<Boolean>>(); // OK
+    }
+
+    public class MyClass<Foo extends Boolean> {} // OK
+
+}
+
+class Bounded {
+
+    public static <Boolean> void foo() {} // warn
+
+}
+
+class Foo<T extends Boolean & Serializable> { // OK
+
+    void foo() {}
+
+}
+
+@interface Annotation {
+
+    Class<? extends Boolean>[] nonPublic(); // OK
+    public Class<? extends Boolean>[] value() default Boolean.class; // warn
```

Comment:
Violation is on `Class` portion. Why isn't there a violation on `default`?

Reply:
Looks like a bug. There should be exactly one violation. For the `default` field.

Reply:
Everything is OK here. The test is set up as follows:
```
 * This test-input is intended to be checked using following configuration:
 *
 * illegalClassNames = {Boolean, Foo, Serializable}
 * memberModifiers = {LITERAL_PUBLIC, FINAL}
```
So,
```
Class<? extends Boolean>[] nonPublic(); // OK: nonPublic is non public nor final
public Class<? extends Boolean>[] value() default Boolean.class; // warn: value() is public, so it should be checked
```

Reply:
Yes, but it only violates on `Class<? extends Boolean>` while not on `default Boolean.class`.
If you did `public Class<?>[] value() default Boolean.class;` there would be no violation.

I think its just another case of something outside the scope of this method.